salt/states/http.py

```diff
@@ -123,7 +123,7 @@ def query(name, match=None, match_type='string', status=None, status_type='strin
 
     if status is not None:
         if status_type == 'string':
-            if data.get('status', '') == str(status):
+            if str(data.get('status', '')) == str(status):
```

Comment:
@mattp- Can we use `salt.utils.stringutils.to_str_here`?

Reply:
this actually is a bit of an issue as status can be an int which to_str seems to not coerce. does it make sense to use ?

Reply:
@mattp- Yes, makes sense.